src/runtime/GlobalObjectBuiltinArray.cpp

```diff
@@ -1846,6 +1846,25 @@ static Value builtinArrayIteratorNext(ExecutionState& state, Value thisValue, si
     return iter->next(state);
 }
 
+static Value builtinArrayAt(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    // Let O be ? ToObject(this value).
+    if (!thisValue.isObject()) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, state.context()->staticStrings().ArrayIterator.string(), true, state.context()->staticStrings().next.string(), ErrorObject::Messages::GlobalObject_CalledOnIncompatibleReceiver);
+    }
+    Object* obj = thisValue.asObject();
+    size_t len = obj->length(state);
+    double relativeStart = argv[0].toInteger(state);
+    if (relativeStart < 0) {
+        relativeStart = len + relativeStart;
+    }
+    if (relativeStart < 0 || relativeStart >= len) {
+        return Value();
+    }
+
+    return obj->get(state, ObjectPropertyName(state, Value(relativeStart).toString(state))).value(state, thisValue);
```

Comment:
I think that we can simply call `getIndexedProperty` here because property name is an integer value.